pageserver/src/http/routes.rs

```diff
@@ -1129,6 +1146,46 @@ async fn timeline_download_remote_layers_handler_get(
     json_response(StatusCode::OK, info)
 }
 
+async fn deletion_queue_flush(
+    r: Request<Body>,
+    cancel: CancellationToken,
+) -> Result<Response<Body>, ApiError> {
+    let state = get_state(&r);
+
+    if state.remote_storage.is_none() {
+        // Nothing to do if remote storage is disabled.
+        return json_response(StatusCode::OK, ());
+    }
+
+    let execute = parse_query_param(&r, "execute")?.unwrap_or(false);
+
+    tokio::select! {
+        flush_result = async {
+            if execute {
+                state.deletion_queue_client.flush_execute().await
+            } else {
+                state.deletion_queue_client.flush().await
+            }
+        } => {
```

Comment:
rustfmt cannot format within macros, but even if it did this would be quite confusing.

```suggestion
    let flush = async {
        if execute {
            state.deletion_queue_client.flush_execute().await
        } else {
            state.deletion_queue_client.flush().await
        }
    };

    tokio::select! {
        flush_result = flush => {
```

`flush_result` cannot be anything other than the result from the `flush` future, feels extra, would rename that to `res` or whatever.

If the `_ = cancel.cancelled() => { ... }` would have reset, could early return, and then handle the `res` (or `flush_result`) in the top level body which would gain formatting as well.

I think the cancellation token here means that the request went away OR shutdown. If you only want shutdown then it should come from within frontend/backend of which I don't really understand the meaning yet.

Reply:
I've refactored this in 7b05ec28250afb8003be86a64e6fbb3249578789 to separate out the flush part from the select part.

> I think the cancellation token here means that the request went away OR shutdown. 

Yes, but if the request went away then our ApiError is going nowhere.  So it makes sense to yield that as our response: if that response is going to a client, then the shutdown status will be accurate, otherwise it goes nowhere.

We could call it something other than ShuttingDown in ApiError, but I think the logic would be the same.